src/berry_mill/imgdescr/loader.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+from typing import Any
+import lxml.etree as ET
+from berry_mill.imgdescr.descr import ApplianceDescription
+
+class UqList(list):
+    """
+    List that acts like a Set, skipping elements that already exist.
+    """
+    def append(self, __object: Any) -> None:
+        if __object not in self:
+            super().append(__object)
+
+
+class Loader:
+    def __init__(self) -> None:
+        self.__i_stack = UqList()
+
+    def _traverse(self, pth: str) -> None:
+        """
+        Traverse the inheritance path
+        """
+        doc: ET.Element|None = None
+        try:
+            with open(pth) as fp:
+                doc = ET.fromstring(fp.read().encode("utf-8"))
+        except Exception as exc:
+            raise IOError(f"Exception while accessing \"{pth}\": {exc}")
+
+        l_iht:list[ET.Element]|None = ApplianceDescription.find_all("inherit", doc)
+        if l_iht:
+            self.__i_stack.append(next(iter(l_iht)).attrib["path"])
+            self._traverse(self.__i_stack[-1])
+        else:
+            self.__i_stack.append(pth)
+
+    def _flatten(self) -> str:
+        """
+        Flatten traversal path
+        """
+        self.__i_stack.reverse()
+
+        # init base
+        descr:ApplianceDescription|None = None
+        with open(next(iter(self.__i_stack))) as fd:
+            descr = ApplianceDescription(fd.read())
+
+        for pth in self.__i_stack[1:]:
+            with open(pth) as fp:
+                descr = ApplianceDescription(fp.read(), descr.to_str())
+
+        return descr.to_str()
+
+    def load(self, pth: str) -> str:
```

Comment:
But that has nothing to do with it. User defines the tree. Best if user does something like `path="../base.xml"` or `path="base.xml"` etc. Which means you first render the tree, then grab the whole output as _one single XML file_, dump it elsewhere, copypaste all root/cfg stuff, cd there, and then build the appliance as usual Kiwi does.

Reply:
Surely, you need to render the tree from the correct directory point, hence you `cd` to that inherited appliance _first_, and then render, and then move elsewhere you need... That is the only limitation. But that is the same as `Makefile` would work or any compiler: as long as it finds the root src, it will then resolve it. That part of the integration is outside of the scope of Xtree class.